src/modm/math/geometry/vector.hpp

```diff
@@ -11,141 +12,552 @@
+	requires std::convertible_to<U, T>
+	void assign(U v, Args... args)
+	{
+		if constexpr(std::integral<T> && std::floating_point<U>)
+			*(this->begin() + K) = std::round(v);
+		else
+			*(this->begin() + K) = v;
+
+		assign<K + 1>(args...);
+	}
+
+ 	template<std::size_t K, typename U, std::size_t M, class... Args>
+	// TODO One variadic method for modm::Vector and std::array
+	// requires std::derived_from<..., std::array>
+	void assign(const Vector<U, M> &vec, Args... args)
+	{
+		if constexpr(std::integral<T> && std::floating_point<U>)
+			std::transform(vec.begin(), vec.end(), this->begin() + K, [](U v) { return std::round(v); });
+		else
+			std::copy(vec.begin(), vec.end(), this->begin() + K);
+
+		assign<K + M>(args...);
+	}
+
+	template<std::size_t K, typename U, std::size_t M, class... Args>
+	void assign(const std::array<U, M> &arr, Args... args)
+	{
+		if constexpr(std::integral<T> && std::floating_point<U>)
+			std::transform(arr.begin(), arr.end(), this->begin() + K, [](U v) { return std::round(v); });
+		else
+			std::copy(arr.begin(), arr.end(), this->begin() + K);
+
+		assign<K + M>(args...);
+	}
+
+public:
+	constexpr Vector()
+	{ this->fill(0); }
 
 	/**
-	 * \brief	Class for handling common point operations
+	 * @brief 			Multi purpose constructor.
 	 *
-	 * Basic data type of all geometric operations. Used to represent vectors
-	 * as well as particular points in the coordinate system.
-	 *
-	 * \section	point_vector	Point vs. vector
+	 * @param args		scalars, Vectors and Arrays in your desired order
+	 */
+	template <class... Args>
+	constexpr Vector(Args... args)
+	{ assign<0>(args...); }
+
+	template <typename U>
+	requires std::convertible_to<U, T>
+	constexpr explicit Vector(U v)
+	{
+		if constexpr(std::integral<T> && std::floating_point<U>)
+			this->fill(std::round(v));
+		else
+			this->fill(v);
+	}
+
+	template <typename U>
+	constexpr Vector(const U (&arr)[N])
+	{
+		if constexpr(std::integral<T> && std::floating_point<U>)
+			std::transform(std::begin(arr), std::end(arr), this->begin(), [] (U v) { return std::round(v); });
+		else
+			std::copy(std::begin(arr), std::end(arr), this->begin());
+	}
+
+	// TODO matrix constructor
+	/* constexpr Vector(const Matrix<T, N, 1> &rhs)
+	{ std::copy(std::begin(rhs.element), std::end(rhs.element), this->begin()); } */
+
+	// matrix assignment
+	/* constexpr Vector& operator= (const Matrix<T, N, 1> &rhs)
+	{
+		std::copy(std::begin(rhs.element), std::end(rhs.element), this->begin());
+		return *this;
+	} */
+
+	// accessors for x, y, z, w, v, u
+	constexpr T x() const { return this->operator[](0); }
+	constexpr T& x() { return this->operator[](0); }
+
+	constexpr T y() const requires (N > 1) { return this->operator[](1); }
+	constexpr T& y() requires (N > 1) { return this->operator[](1); }
+
+	constexpr T z() const requires (N > 2) { return this->operator[](2); }
+	constexpr T& z() requires (N > 2) { return this->operator[](2); }
+
```

Comment:
!!! API-change: Vector components x, y, z, w are no more plain properties but methods. It's obligatory for the single-header rewrite, see @chris-durand _undefined behaviour_ comment above.